### PR TITLE
Enhances Inline Assistant by adding Comprehensive Context Awareness

### PR DESCRIPTION
**Pull request by catrielmuller**
## Context

This PR enhances the Ghost inline assistant by adding comprehensive context awareness through AST (Abstract Syntax Tree) integration, document tracking, and improved code understanding capabilities.

## Implementation

- AST Integration: Adds tree-sitter-based AST parsing to understand code structure and provide better context-aware suggestions
- Enhanced Context System: Implements comprehensive context gathering including recent operations, diagnostics, and open files
- Improved Response Parsing: Updates the strategy to handle both diff-based and full-content response formats from the LLM

## How to Test

Press Cmd+I / Cmd+L on any document.

## Summary by CodeRabbit

* **New Features**
  * Enhanced Inline Assistant with comprehensive context awareness, including recent user actions and document diagnostics.
  * Added advanced context enrichment using abstract syntax tree (AST) analysis for more accurate code suggestions.
  * User actions (additions, deletions, modifications, refactors, formatting) are now detected and summarized for improved suggestion relevance.
  * Integrated automatic document lifecycle tracking with AST parsing for up-to-date context.
  * Added a new keybinding to apply all suggestions with Shift+Tab.

* **Improvements**
  * Visual decorations for code changes now use thinner borders for a more refined appearance.
  * Suggestion prompts and responses now include richer context, such as AST and diagnostics.
  * Improved prompt clarity distinguishing addition/modification and deletion scenarios.
  * Enhanced response parsing supporting multiple formats for better suggestion accuracy.
  * Cleaner handling of suggestion cancellation and state resets.

* **Bug Fixes**
  * Improved handling of edge cases when applying or navigating suggestions.

* **Tests**
  * Added and updated comprehensive test suites covering document storage, AST parsing, user action detection, and prompt generation.
  * Refactored test cases for better normalization and coverage of new context features.

* **Documentation**
  * Added and updated metadata and descriptions for new features and changes.

### CODE REVIEW

**Review by chrarnoldus — APPROVED** [review]
but please take a look at https://github.com/Kilo-Org/kilocode/issues/1465#issuecomment-3117799514

File: .changeset/every-months-dance.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"kilo-code": minor
+---
+
+Enhances Inline Assistant by adding Comprehensive Context Awareness
```
Comment (hassoncs) [inline .changeset/every-months-dance.md]:
```suggestion
Improve Inline Assist suggestions by adding comprehensive context awareness
```

**Review by hassoncs — APPROVED** [review]
Nice! it's working really well! There's some code we can DRY up GhostStrategy, but let's merge it and iterate from there

**Review by hassoncs — APPROVED** [review]
